docs/configuration/initialization.md

```diff
@@ -0,0 +1,129 @@
+}
+```
+
+The `...` section will still be your [options](/configuration/options), but you now have the possibility to execute/modify certain things on the request.
+
+You could for example log the request, add headers, read `query` or `body` parameters, whatever you would do in an API route.
+
+:::tip
+Since this is a catch-all route, remember to check what kind of NextAuth.js "action" is running. Compare the REST API with the `req.query.nextauth` parameter.
+
+For example to execute something on the "callback" action when the request is a POST method, you can check for `req.query.nextauth.includes("callback") && req.mehod === "POST"`
+:::
+
+:::note
+`NextAuth` will implicitly close the response (by calling `res.end`, `res.send` or similar), so you should not run code **after** `NextAuth` in the function body. Using `return NextAuth` makes sure you don't forget that.
+:::
+
+Any variable you create this way will be available in the `NextAuth` options as well, since they are in the same scope.
+
+```ts title="/pages/api/auth/[...nextauth].ts"
+import type { NextApiRequest, NextApiResponse } from "next"
+import NextAuth from "next-auth"
+
+export default async function auth(req: NextApiRequest, res: NextApiResponse) {
+
+  if(req.query.nextauth.includes("callback") && req.method === "POST") {
+    console.log(
+      "Handling callback request from my Identity Provider",
+      req.body
+    )
+  }
+
+  // Get a custom cookie value from the request
+  const someCookie = req.cookies["some-custom-cookie"]
+
+  return await NextAuth(req, res, {
+    ...
+    callbacks: {
+      session({ session, token }) {
+        // Return a cookie value as part of the session
+        // This is read when `req.query.nextauth.includes("session") && req.method === "GET"`
+        session.someCookie = someCookie
+        return session
+      }
+    }
+  })
+}
+```
+
+A practical example could be to not show a certain provider on the default sign-in page, but still be able to sign in with it. (The idea is taken from [this discussion](https://github.com/nextauthjs/next-auth/discussions/3133)):
+
+```js title="/pages/api/auth/[...nextauth].js"
+import NextAuth from "next-auth"
+import CredentialsProvider from "next-auth/providers/credentials"
+import GoogleProvider from "next-auth/providers/google"
+
+export default async function auth(req, res) {
+  const providers = [
+    CredentialsProvider(...),
+    GoogleProvider(...),
+  ]
+
+  const isDefaultSigninPage = req.method === "GET" && req.query.nextauth.includes("signin")
+
+  // Will hide the `GoogleProvider` when you visit `/api/auth/signin`
+  if (isDefaultSigninPage) providers.pop()
+
+  return await NextAuth(req, res, {
+    providers,
+    ...
+  })
+}
+```
+
+To completely understand what actions correspond to and what methods are supported, check out the [REST API documentation](/getting-started/rest-api), or look at the [source code](https://github.com/nextauthjs/next-auth/blob/beta/src/core/index.ts).
```

Comment:
Maybe simplify this as well, something along the lines of - "For more details on all available actions and which methods are supported, please check out the [REST API Documentation]() or the appropriate area in [the source code]()".